packages/cln_plugin/lib/src/cln_plugin_base.dart

```diff
@@ -1,6 +1,47 @@
 // TODO: Put public facing types in this file.
 
-/// Checks if you are awesome. Spoiler: you are.
-class Awesome {
-  bool get isAwesome => true;
+import 'dart:io';
+
+import 'icln_plugin_base.dart';
+
+class Plugin implements ICLNPlugin {
+  // Data members
+  /*
+  ManifestMethod _manifest;
+  List<Interceptor> _preInterceptor;
+  List<Interceptor> _postInterceptor;
+  Map<String, Object> parameters;
+  bool _parametersReady;
+  JsonConverter converter;
+  CLightningPluginConfig configs;
+   */
+
+  Plugin() {
+    // Data member initializations
+  }
+
+  @override
+  void start() {
+    // This function will communicate with the rpc.
+    print("The plugin is started.");
+    try {
+      String? messageSocket;
+      while((messageSocket = stdin.readLineSync()) != null) {
+        // Already checked is stdin is not null, why trim and check again??
+        if(messageSocket!.trim().isEmpty) {
+          continue;
+        }
+        // TODO: Log messageSocket to stdout as a debug message
+        int deep = 0;
+
+        // Dart doesn't have a max 32 bit int, and casting a double would return errors for some reason, hence the hardcode value.
+        final int maxDeep = 1073741824;
+
+        // TODO: Error handling for JSON and max recursion depth errors
+      }
+    } catch (error, stacktrace) {
+      print('$error:$stacktrace');
+      //TODO: log this error
+    }
+  }
```

Comment:
This is something that can be inserted into the abstract class, the abstract class can contain method implementation, the Java interface is no!

The basic idea that an abstract class need to follow is composed of the following criteria:
- The method has an implementation that doesn't depend on the user? So it needs to go in the abstract class, and we can hide this from the developer.
-  The method need a user definition? so need to be declared without body, and the user must implement it.